### PR TITLE
Fix: can't link assistant to Slack channel if Slack channel is not in…

### PR DESCRIPTION
**Pull request by lasryaric**
… our DB

### CODE REVIEW

File: connectors/src/api/slack_channels_linked_with_agent.ts [hunk 1]
```diff
@@ -66,19 +68,39 @@ const _patchSlackChannelsLinkedWithAgentHandler = async (
...
+      const remoteChannelsById = remoteChannels.reduce((acc, ch) => {
+        acc[ch.id] = ch;
+        return acc;
+      }, {} as Record<string, { id: string; name: string }>);
+      const createdChannels = await Promise.all(
+        missingSlackChannelIds.map((slackChannelId) => {
+          const remoteChannel = remoteChannelsById[slackChannelId];
+          if (!remoteChannel) {
+            throw new Error(
+              `Unexpected error: Access to the Slack channel ${slackChannelId} seems lost.`
```
Comment (PopDaph) [inline connectors/src/api/slack_channels_linked_with_agent.ts:85]: If we thow no channel will be attached to the agent and the error is managed from Front?

File: connectors/src/api/slack_channels_linked_with_agent.ts [hunk 2]
```diff
@@ -98,6 +120,27 @@ const _patchSlackChannelsLinkedWithAgentHandler = async (
       )
     );
   });
+  const joinPromises = await Promise.all(
+    slackChannelIds.map((slackChannelId) =>
+      joinChannel(parseInt(connectorId), slackChannelId)
```
Comment (PopDaph) [inline connectors/src/api/slack_channels_linked_with_agent.ts:125]: So the Bot joins automatically a channel if the channel is selected in the list to be able to answer but it won't index the content of this channel?

**Review by PopDaph — APPROVED** [review]
I would have expected that we don't show a list of channels to the user that we don't know as it feels like a hack but discussed IRL and you explain me why this is what we want (we want to add the SlackChannel on db only when we need it 👍🏻 )

File: connectors/src/api/slack_channels_linked_with_agent.ts (hunk 2 above)
Comment (lasryaric) [reply connectors/src/api/slack_channels_linked_with_agent.ts:125]: Correct.

**Pull request merged**